### DISCUSSION TITLE
How to escape special characters in path regex?

### DISCUSSION THREAD

**Discussion by privatenumber** [community/community#10031] [Other Feature Feedback, Questions, & Ideas] [answered]
Trying to find all `.d.ts` files, and I started with [query `path:/.d.ts$/`](https://cs.github.com/?scopeName=All+repos&scope=&q=path%3A%2F.d.ts%24%2F) but realized `.` was being treated as a wildcard.

I tried escaping them with a backslash but it returns no results with [query `path:/\.d\.ts$/`](https://cs.github.com/?scopeName=All+repos&scope=&q=path%3A%2F%5C.d%5C.ts%24%2F).

How can I escape special characters in the regex pattern?

Edit:
I think the backslash is how you escape them (based on the examples on the landing page), however, it's strange there are no results. Perhaps `.d.ts` files aren't indexed?

**Comment by colinwm** [comment, marked as answer]
Your query is correct. We had a bug with our indexer which caused it to ignore all `*.d.ts` files. Although this bug is now fixed, you'll have to wait for us to reindex everything to see them again, which should probably take ~24h or so. Sorry about that!
